### DISCUSSION TITLE
Avalonia no ViewModelLocator

### DISCUSSION THREAD

**Discussion by flarive34rs** [AvaloniaUI/Avalonia#6687] [Q&A]
Hi,

I'm have just created a new Avalonia 0.10.7 new project with VS2019 on Windows.

A ViewLocator was automatically added to the newly crreated project, it's great !
I'm used to use also a ViewModelLocator to access the viewmodels from anywhere.

I can't find any ViewModelLocator class already made dealing with ReactiveObject base view models.

How can i create a ViewModelLocator for my Avalonia project 
Does any body would have an existing ViewModelLocator class ready to use please ?

Thanks a lot

**Comment by maxkatz6** [comment]
I believe ViewModelLocator depends more on used IoC than on ReactiveUI or Avalonia.
I.e. it's basically a wrapper on top of something like Autofac or Microsoft.Extensions.DependencyInjection. I don't have one that I can share, but it shouldn't be hard to write one. Also it doesn't depend on Avalonia, so it can be reused from WPF or XF.
